example/src/main/java/org/secondbase/example/main/HelloAll.java

```diff
@@ -0,0 +1,66 @@
+package org.secondbase.example.main;
+
+import io.prometheus.client.Counter;
+import java.io.IOException;
+import org.secondbase.consul.ConsulModule;
+import org.secondbase.core.SecondBase;
+import org.secondbase.core.SecondBaseException;
+import org.secondbase.core.config.SecondBaseModule;
+import org.secondbase.flags.Flag;
+import org.secondbase.flags.Flags;
+import org.secondbase.logging.JsonLoggerModule;
+import org.secondbase.webconsole.HttpWebConsole;
+import org.secondbase.webconsole.PrometheusWebConsole;
+import org.secondbase.webconsole.widget.Widget;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HelloAll {
+    @Flag(name="variable")
+    private static String var = "default";
+    @Flag(name="counter")
+    private static int counter = 1;
+
+    private static final Logger LOG = LoggerFactory.getLogger(HelloAll.class.getName());
+
+    private HelloAll() {}
+
+    public static void main(final String[] args) throws SecondBaseException, IOException {
+
+        final String[] realArgs = {
+                // Consul settings
+                "--consul-host=localhost:8500",
+                "--service-name=HelloAll",
+                "--service-port=8000",
+                "--service-environment=testing",
+                "--service-health-check-path=/healthz",
+                "--service-tags=tagone,tagtwo",
+                // Logging settings
+                "--service=HelloAll",
+                "--datacenter=local",
+                "--environment=testing",
+                // Webconsole settings
+                "--webconsole-port=8000"
+        };
+
+        // Set up json logger module first, since it can define how the other modules do logging.
+        final SecondBaseModule jsonLogger = new JsonLoggerModule();
+
+        final Counter mycounter = Counter.build("mycounter", "a counter").register();
+
+        final Widget prometheusWidget = new PrometheusWebConsole();
+        final Widget[] widgets = {prometheusWidget};
+
+        final ConsulModule consul = new ConsulModule();
+        final SecondBaseModule webconsole = new HttpWebConsole(widgets);
+        final SecondBaseModule[] modules = {consul, webconsole, jsonLogger};
+
+        final Flags flags = new Flags().loadOpts(HelloAll.class);
+
+        new SecondBase(realArgs, modules, flags);
+
+        mycounter.inc(counter);
+
+        LOG.info(var);
```

Comment:
I'm not sure about the logic here when it comes to enabling consul. I.e. how to use variables properly. If no service name is specified nothing is happening but if no consul is available we get an error.